### PR TITLE
update plan

### PR DESCRIPTION
**Pull request by zyndiecate**
epic https://github.com/giantswarm/giantswarm/issues/108
origin https://github.com/giantswarm/app-service/pull/518

### CODE REVIEW

**Comment by zyndiecate** [comment]
@ewoutp RFR

File: update.go [hunk 1]
```diff
@@ -147,13 +152,16 @@ func updateService(definitionFilePath string) error {
...
 		return maskAny(err)
 	}
 
 	// confirm
 	if !updateAutoAccept {
-		ok, err := user.Confirmf("\nAre you sure to update service \"%s\" based on the given plan? [y|n] ", serviceName)
+		// TODO
+		// if -q msg := "Are you sure you want to proceed with the update?"
+
+		ok, err := user.Confirmf("Are you sure you want to proceed with the update as described above? [y|n] ")
```
Comment (ewoutp) [inline update.go]: Please put the messages in constants.

File: update.go [hunk 2]
```diff
@@ -188,32 +196,130 @@ func updateService(definitionFilePath string) error {
...
-	tableData := []string{}
+	fmt.Println()
+	fmt.Println("--------------------------------------------------------------------------------")
+	fmt.Println()
+
+	numAddedComps := len(updatePlan.AddedComponents)
+	if numAddedComps > 0 {
+		subject := "component"
+		if numAddedComps > 1 {
+			subject += "s"
```
Comment (ewoutp) [inline update.go]: Prefer to make it explicit.
```
subject = "components"
```

File: update.go [hunk 3]
```diff
@@ -188,32 +196,130 @@ func updateService(definitionFilePath string) error {
...
+
+		fmt.Println("--------------------------------------------------------------------------------")
+		fmt.Println()
+	}
+
+	numDeletedComps := len(updatePlan.DeletedComponents)
+	if numDeletedComps > 0 {
+		subject := "component"
+		if numDeletedComps > 1 {
+			subject += "s"
```
Comment (ewoutp) [inline update.go]: `subject = "components"`

File: update.go [hunk 4]
```diff
@@ -188,32 +196,130 @@ func updateService(definitionFilePath string) error {
...
+		fmt.Println("--------------------------------------------------------------------------------")
+		fmt.Println()
+	}
+
+	numWarns := len(updatePlan.MetaInfos.Warnings)
+	if numWarns > 0 {
+		fmt.Println("Warnings")
+		fmt.Println()
+
+		// TODO sort warnings ???
```
Comment (ewoutp) [inline update.go]: Why and if so, in which order? asci, severity,...

Comment (zyndiecate) [inline update.go]: Maybe ordered alphabetically by component names, or warning type. I don't know.

File: update.go (hunk 1 above)
Comment (zyndiecate) [inline update.go]: Done.

File: update.go (hunk 2 above)
Comment (zyndiecate) [inline update.go]: Done.

File: update.go (hunk 3 above)
Comment (zyndiecate) [inline update.go]: Done.

**Comment by zyndiecate** [comment]
Adressed all comments.

File: update.go (hunk 4 above)
Comment (ewoutp) [inline update.go]: For know this should be fine.

**Comment by ewoutp** [comment]
LGTM, let me know when the app-service / api are ready, then I'll do an integration test.

**Comment by marians** [comment]
Reminder: We need to update the test texts to match the usage and command output.

**Comment by zyndiecate** [comment]
CI breaks because of `010-create-invalid.t`. Not scope of this PR.

**Comment by zyndiecate** [comment]
@marians thanks for remembering about the tests. :)

**Pull request merged**